Add Ray standalone app (#1102)
netgroup-polito/CrownLabs: provisioning/standalone/vscode/ray/workspace/README.md

# Connect to the cluster
ray.init(
    address=RAY_HEAD,
    ignore_reinit_error=True
)
```

---

## 🧠 The 3 Core Concepts
Using Ray boils down to three simple steps.

### 1. The Decorator (`@ray.remote`)
To tell Ray that a function should run on the cluster (instead of your local machine), add the `@ray.remote` decorator.
You can also specify exactly what hardware the function needs.

```python
# This task requires exactly 1 GPU to run
@ray.remote(num_gpus=1)
def train_model():
    import torch
    return torch.cuda.is_available()

# This task only needs a CPU core
@ray.remote(num_cpus=1)
def calculate_math():
    return 1 + 1
```

### 2. Submitting Tasks (`.remote()`)
To execute the function on the cluster, do not call it normally (e.g., `train_model()`).
Instead, append `.remote()`.
This immediately sends the task to the cluster queue and returns a "Future" (an Object Reference), allowing your main script to keep running without freezing.

```python
# Submit 10 tasks in parallel
futures = [calculate_math.remote() for _ in range(10)]
```

### 3. Retrieving Results (`ray.get()`)
To get the actual data back from the cluster, pass the Future(s) to `ray.get()`.

> **Note:** This is a blocking operation; your script will pause here until the cluster finishes the work.

Reviewer: ```suggestion
**Note:** This is a blocking operation; your script will pause here until your remote task finishes the work.
```

```python
results = ray.get(futures)
print(results) # Output: [2, 2, 2, 2...]
```

---

## 📂 Sample Scripts

> Inside the `samples/` folder, you will find ready-to-run scripts progressing from basic sanity checks to advanced machine learning sweeps.

Reviewer: ```suggestion
The `samples/` folder contains some ready-to-run scripts progressing from basic sanity checks to advanced machine learning sweeps.
```

> We highly recommend running them in this order:

Reviewer: Names of the scripts are not aligned with the .py files present in this repository.
